src/UserContext/AnonymousRequestMatcher.php

```diff
@@ -13,36 +13,62 @@
 
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Matches anonymous requests using a list of identification headers.
  */
 class AnonymousRequestMatcher implements RequestMatcherInterface
 {
-    private $sessionNamePrefix;
+    /**
+     * The options configured in the constructor argument or default values.
+     *
+     * @var array
+     */
+    protected $options;
 
     /**
-     * @param string $sessionNamePrefix Prefix for session cookies. Must match your PHP session configuration
+     * When creating this matcher, you can configure a number of options.
+     *
+     * - user_identifier_headers: List of request headers that authenticate a non-anonymous request.
+     * - session_name_prefix:     Prefix for session cookies. Must match your PHP session configuration.
+     *
+     * @param array $options Options to overwrite the default options
+     *
+     * @throws \InvalidArgumentException if unknown keys are found in $options
      */
-    public function __construct($sessionNamePrefix)
+    public function __construct(array $options = array())
     {
-        $this->sessionNamePrefix = $sessionNamePrefix;
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults(array(
+            'user_identifier_headers' => array('Cookie', 'Authorization', 'HTTP_AUTHORIZATION', 'PHP_AUTH_USER'),
+            'session_name_prefix' => 'PHPSESSID',
```

Comment:
we always pass values from either the defaults in the listener or the symfony config, right? if thats correct i'd rather make these required and have no defaults to not repeat the defaults in too many places.